src/components/MovieDetail/MovieDetail.js

```diff
@@ -1,65 +1,108 @@
 import React, { Component } from 'react';
 import './MovieDetail.css';
 import { BsPlayFill } from 'react-icons/bs';
-import { getSelectedMovie, handleErrors } from '../../utilities';
+import { getSelectedMovie, handleErrors, getTrailerKey } from '../../APICalls';
+import { formatDate, formatGenres } from '../../utilities';
+import Trailer from '../Trailer/Trailer';
 
 class MovieDetail extends Component {
   constructor(props) {
     super(props);
-    this.state = { ...props };
+    this.state = {
```

Comment:
Well done. This is much more readable now. It's more lines of code but it's much clearer what's getting stored in state.